src/components/calcite-input-time-picker/calcite-input-time-picker.tsx

```diff
@@ -0,0 +1,276 @@
+import {
+  Component,
+  Element,
+  Host,
+  VNode,
+  h,
+  Prop,
+  Listen,
+  Event,
+  EventEmitter,
+  State,
+  Method
+} from "@stencil/core";
+import { guid } from "../../utils/guid";
+import { parseTimeString, Time, validateTimeString } from "../../utils/time";
+import { Theme } from "../interfaces";
+
+@Component({
+  tag: "calcite-input-time-picker",
+  styleUrl: "calcite-input-time-picker.scss",
+  scoped: true
+})
+export class CalciteInputTimePicker {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteInputTimePickerElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** The disabled state of the time input */
+  @Prop({ reflect: true }) disabled?: boolean = false;
+
+  /** Format of the hour value (12-hour or 24-hour) (this will be replaced by locale eventually) */
+  @Prop({ reflect: true }) hourDisplayFormat: "12" | "24" = "12";
+
+  /** The name of the time input */
+  @Prop({ reflect: true }) name?: string;
```

Comment:
nitpick: does name need to be reflected?

Reply:
Don't we normally reflect name attributes?  The name gets passed onto the calcite-input which is then subsequently recognized by parent forms, and because this custom element is not using any form-associated API, should be harmless, although I can see the argument in not reflecting it if it is always going to be included on the input anyway; that way queries for the name won't turn up more than one result.